File: src/slides/go-duck/dumbdb.go
package main

import (
	"os"
	"net/http"
	"path/filepath"
	"io/ioutil"
	"encoding/json"
	"strconv"
)

var DIR string

func getContent(name string) map[string]string {
	data, err := ioutil.ReadFile(filepath.Join(DIR, name))
    if err == nil {
    	return map[string]string{"content": string(data)}
    }
    return map[string]string{"error": "data not found"}
}

func Responder(w http.ResponseWriter, req *http.Request) {
	name := req.FormValue("id")
	data, _ := json.Marshal(getContent(name))
	w.Write(data)
}

func main() {
	if len(os.Args) == 1 {
		println("Supply path to directory with data as first argument")
		return
	}
	DIR = os.Args[1]

	http.HandleFunc("/", Responder)
	http.ListenAndServe(":8081", nil)
}

File: src/slides/go-duck/dumbdb.py
#!/usr/bin/env python

import sys
import os
import json
from tornado import ioloop, web

DIR = None

class Responder(web.RequestHandler):
    def get(self):
        name = self.get_argument('id')
        try:
            with file(os.path.join(DIR, name)) as f:
                content = f.read()
                self.write(json.dumps({"content": content}))
        except IOError:
            self.write(json.dumps({"error": "data not found"}))

app = web.Application([(r'/', Responder)])

if __name__ == '__main__':
    try:
        DIR = sys.argv[1]
    except IndexError:
        print "Supply path to directory with data as first argument"
        sys.exit()

    app.listen(8080, 'localhost')
    ioloop.IOLoop.instance().start()
